docs/source/teams/pluggable_auth.md

```diff
@@ -0,0 +1,668 @@
+
+(internal-mode)=
+
+**Introduction to Internal Mode**
+
+Internal Mode eliminates the need for Auth0, thereby removing FiftyOne Team's
+dependency on any external services. This mode does not require external
+connectivity, making it suitable for environments where security is paramount
+or internet access is limited or not allowed. Unlike Legacy Mode, Internal Mode
+operates on an immediate consistency basis, ensuring that changes are reflected
+across FiftyOne Teams instantly. Directory data is immediately written to
+MongoDB, and organizations have the autonomy to manage their Identity Provider
+Configuration. **NOTE: SAML support is not available in Internal Mode.**
+
+(identity-providers)=
+
+## Identity Providers
+
+In Internal Mode you can use either the CAS REST API or Super Admin UI to
+configure FiftyOne teams to authenticate users directly against OIDC or OAuth
+compatible Identity Providers. Below is an example configuration for KeyCloak
+as an Identity Provider.
+
+```json
+{
+    "id": "keycloak-example",
+    "wellKnown": "https://keycloak.dev.acme.ai/auth/realms/acme/.well-known/openid-configuration",
+    "name": "KeyCloak",
+    "type": "oauth",
+    "authorization": {
+        "url": "https://keycloak.acme.ai/auth/realms/acme/protocol/openid-connect/auth",
+        "params": {
+            "scope": "openid email profile"
+        }
+    },
+    "clientId": "...",
+    "clientSecret": "...",
+    "issuer": "https://keycloak.dev.acme.ai/auth/realms/acme",
+    "token": {
+        "url": "https://keycloak.dev.acme.ai/auth/realms/acme/protocol/openid-connect/token"
+    },
+    "userinfo": {
+        "url": "https://keycloak.dev.acme.ai/auth/realms/acme/protocol/openid-connect/userinfo"
+    },
+    "idToken": true,
+    "style": {
+        "logoDark": "https://images.com/keycloak.svg",
+        "bg": "#47abc6",
+        "text": "#fff"
+    },
+    "profile": "(e,l)=>(console.log({profile:e,tokens:l}),{id:e.sub,name:e.name,email:e.email,image:null})",
+    "allowDangerousEmailAccountLinking": true
+}
+```
+
+<!--
+**Migrating from Legacy to Internal Mode**
+
+The migration from Legacy to Internal Mode begins with configuring an
+authentication provider through CAS (UI or REST). Following this, all existing
+user data must be migrated from Auth0, utilizing the management SDK or Auth0
+API to ensure a comprehensive transfer of information.
+
+For each user, the migration involves several steps:
+
+1. Creating a FiftyOne Teams user profile via `POST /cas/api/users`.
+2. Assigning the user to the default organization through a membership entry.
+3. Linking the user's account to the new authentication provider by creating an
+   account reference via `POST /cas/api/accounts`.
+
+The final step in the migration involves changing the `FIFTYONE_AUTH_MODE`
+setting from legacy to internal. This change officially activates Internal
+Mode, completing the migration process.
+-->
+
+## Getting started with Internal Mode
```

Comment:
Maybe we add a note here about invitations not working yet. but saying you can still manage users through the super admin dashboard and rest api / js hooks.